Select a line, show what tests cover it in a RHS list (right-sidebar)

LinkaiQi opened the pull request:
Refer to Issue #20

Features:
1. User can use file-view to select line
2. Show tests that cover the selected line in the RHS list

Try: http://localhost:3000/file?revision=a0334f789772&path=/accessible/atk/Platform.cpp
( Lines 136, 219, 264, 310 .... have test coverages)

LinkaiQi committed:
Add fileviewer sidebar, selected line state, onClick handler function

LinkaiQi committed:
refine sidebar view, add selected state to line componment

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -49,12 +54,43 @@ export default class FileViewerContainer extends Component {
+    console.log(data.length);
+    var stat = []
+
+    for (var i = 0; i < data.length; i++) {
```
Please use `forEach`, like so:

    data.forEach((d, i) => {  
       var coveredLines = d.source.file.covered;

you may not even need `i`

klahnakoski commented on src/components/fileviewercov.js:
```diff
@@ -0,0 +1,55 @@
+    }
+
+  const testItems = testList.map((testNum) => {
+    const test = coverage.data[testNum];
```
You are storing the `testNum` in `testsPerLines`.  You can store the object directly, rather than an index to the object.

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -49,12 +54,43 @@ export default class FileViewerContainer extends Component {
+        if (!stat[line]) {
+          stat[line] = [];
+        }
+        stat[line].push(i);
```
Instead of `stat[line].push(i)`  use `stat[line].push(d)` then

    const testItems = testList.map((testNum) => {
      const test = coverage.data[testNum];
      const testName = test.run.name;
      return <li key={testName}>{testName}</li>
    });

will read

    const testItems = testList.map((test) => {
      return <li key={test.run.name}>{test.run.name}</li>
    });

which then suggests we inline this one-line function

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -49,12 +54,43 @@ export default class FileViewerContainer extends Component {
+    for (var i = 0; i < data.length; i++) {
+      var coveredLines = data[i].source.file.covered;
+      for (var j = 0; j < coveredLines.length; j++) {
+        var line = coveredLines[j];
```
Another example where `forEach` is better.

klahnakoski commented:
This looks great!  Please make the code changes I suggested, and I can merge.

LinkaiQi committed:
Addressing PR/#27 review feedback

1. Using 'forEach' function to iterate array
2. Store 'test' obj directly, rather than an index to it

klahnakoski commented on src/components/fileviewer.js:
```diff
@@ -66,16 +66,16 @@ export default class FileViewerContainer extends Component {
-        var line = coveredLines[j];
+    data.forEach(d => {
+      const coveredLines = d.source.file.covered;
+      coveredLines.forEach(line => {
```
Next time, this could be one line `d.source.file.covered.forEach(line => {`

LinkaiQi commented:
Thanks Kyle, the code is updated.
